src/modes/summary/Overview.svelte

```diff
@@ -27,33 +25,56 @@
   $: DEATHS = new SensorParam($defaultDeathSensor, $metaDataManager);
   $: HOSPITAL_ADMISSION = new SensorParam($defaultHospitalSensor, $metaDataManager);
 
-  function fetchFallback(fetcher, sensor, region, trend) {
-    return trend.then((t) => {
-      if (t && t.value != null) {
-        return t;
-      }
-      return fetcher.fetch1Sensor1Region1DateTrend(sensor, region, DateParam.box(sensor.timeFrame.max));
-    });
-  }
-
   $: trends = fetcher.fetchNSensors1Region1DateTrend([CASES, HOSPITAL_ADMISSION, DEATHS], region, date);
   $: casesTrend = trends[0];
-  $: hospitalTrend = fetchFallback(fetcher, HOSPITAL_ADMISSION, region, trends[1]);
+  $: hospitalTrend = trends[1];
   $: deathTrend = trends[2];
-</script>
 
-<IndicatorWarning sensor={CASES} {date} {region} {fetcher} />
-<IndicatorAnnotations {date} {region} sensor={CASES} range="sparkLine" />
+  let minMaxDate = new Date();
+
+  onMount(() => {
+    [CASES, HOSPITAL_ADMISSION].map((s) => {
+      if (s.timeFrame.max < minMaxDate) {
+        minMaxDate = s.timeFrame.max;
+      }
+    });
+    let urlSearchParams = new URLSearchParams(window.location.search);
+    if (!urlSearchParams.has('date')) {
+      // if no date is specified in the URL, default to the latest day before today with data from all 3 highlighted indicators
+      date.set(minMaxDate);
+    }
+  });
+
+  // warningType is indicator of which exact warning message should be shown.
+  // By default, when user opens page with no specified date, the date will be set to the latest date we have data for all 3 indicators.
+  // In this case, warningType should be set to 1.
+  // In case selected date is set to future date (date > minMaxDate, where we don't have recent data for all 3 indicators), the warningType will be set to 2
+  // which has different warning message.
+  // In case selected date is set to some date which is < minMaxDate, the warningType will be set to 0 which means that we will not show
+  // any warning message.
+
+  // warningType should be set in beforeUpdate() method, to guess correct warningType.
+
+  let warningType = 1;
+  beforeUpdate(() => {
+    if (date.value > minMaxDate) {
+      warningType = 2;
+    } else if (date.value < minMaxDate) {
+      warningType = 0;
+    } else {
+      warningType = 1;
+    }
+  });
```

Comment:
can we do the `date.value != minMaxDate` comparisons inside `NoRecentDataWarning.svelte`?  That way we wouldn't have to pass `warningType`...  Or do we need to do the comparisons inside of the `beforeUpdate()` event here?

Reply:
i talked to Dmytro on slack, he is going to look at this but we will release the current state in the meantime as the functionality is doing what its supposed to.